lib/pbench/test/unit/common/test_config.py

```diff
@@ -1,68 +1,252 @@
-        config = PbenchConfig(_config_path_prefix / "logdir.cfg")
-        assert (
-            config.log_dir == "/srv/log/directory"
-        ), f"Unexpected log directory, {config.log_dir!r}"
-
-    def test_logger_type_provided(self):
-        config = PbenchConfig(_config_path_prefix / "hostport.cfg")
-        assert (
-            config.logger_type == "hostport"
-        ), f"Unexpected logger type, {config.logger_type!r}"
-        assert (
-            config.logger_host == "logger.example.com"
-        ), f"Unexpected logger host value, {config.logger_host!r}"
-        assert (
-            config.logger_port == "42"
-        ), f"Unexpected logger port value, {config.logger_port!r}"
-
-    def test_logger_type_hostport_missing(self):
-        with pytest.raises(BadConfig):
-            PbenchConfig(_config_path_prefix / "hostport-missing.cfg")
-        with pytest.raises(BadConfig):
-            PbenchConfig(_config_path_prefix / "hostport-missing-port.cfg")
+    def mock_file_list(self, cfg_name: str) -> List[str]:
+        """file_list mock module"""
+        return [cfg_name]
+
+    def test_default_logger(self, monkeypatch):
+        monkeypatch.setattr(configtools, "file_list", self.mock_file_list)
+        monkeypatch.setattr(ConfigParser, "__init__", MockedConfigParser.__init__)
+        monkeypatch.setattr(ConfigParser, "get", MockedConfigParser.get)
+        config = PbenchConfig("pbench.cfg")
+        assert config.logger_type == "devlog"
+
+    def test_logger(self, monkeypatch):
+        """verify logger_type value from config file"""
+
+        class mock_MockedConfigParser(MockedConfigParser):
+            def __init__(self):
+                self._sections = {"logging": {"logger_type": "file"}}
+
+        monkeypatch.setattr(configtools, "file_list", self.mock_file_list)
+        monkeypatch.setattr(ConfigParser, "__init__", mock_MockedConfigParser.__init__)
+        monkeypatch.setattr(ConfigParser, "get", MockedConfigParser.get)
+        config = PbenchConfig("pbench.cfg")
+        assert config.logger_type == "file"
+
+    def test_logger_type_hostport(self, monkeypatch):
+        """verify logger_type value as `hostport`, it expects `host` and `port`"""
+
+        class mock_MockedConfigParser(MockedConfigParser):
+            def __init__(self):
+                self._sections = {"logging": {"logger_type": "hostport"}}
+
+        monkeypatch.setattr(configtools, "file_list", self.mock_file_list)
+        monkeypatch.setattr(ConfigParser, "__init__", mock_MockedConfigParser.__init__)
+        monkeypatch.setattr(ConfigParser, "get", MockedConfigParser.get)
+        with pytest.raises(BadConfig) as exc:
+            config = PbenchConfig("pbench.cfg")
+            assert config.logger_type == "hostport"
+            assert exc == "No option 'logger_host' in section: 'logging'"
+
+    def test_logger_type_host_value(self, monkeypatch):
+        """verify `host` value of logger_type `hostport`"""
```

Comment:
This should be based more on the docstring from `test_logger_type_hostport` distinguishing that in this case the key is present but has an empty value.

(The test names could make this distinction more clear, as well, although it's best not to make them too long!)

Reply:
As a general rule of thumb, the docstring for a test should say something similar to "Show that when XXXX happens, function YYYY responds with ZZZZ".  So, for this test, it should be something like, "Show that PbenchConfig raises ValueError when the configuration contains an empty string for 'logger_host'."